src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/PersistorBase.java

```diff
@@ -122,14 +134,26 @@ public synchronized void openNewDBFile() throws SQLException {
     createNewDSLContext();
   }
 
+  /**
+   * This method check if there is a need to delete old sqlite files and create a new one.
+   * Ideally we will be using new sqlite files at the start of every hour, ideally the whenever the
+   * function write is called for the first time in that very hour
+   */
   public synchronized void rotateDBIfRequired() throws ParseException, SQLException {
-    Date currTimeMs = new Date(System.currentTimeMillis());
-    if ((currTimeMs.getTime() - this.fileCreateTime.getTime()) / 1000 > FILE_ROTATION_PERIOD_SECS) {
+    LocalDateTime currentLocalDateTime = getLocalDateTimeFromDateObj(new Date(System.currentTimeMillis()));
```

Comment:
The default constructor takes the current time `new Date()`

Reply:
Done